ntbs-service/DataMigration/NotificationImportService.cs

```diff
@@ -183,6 +183,28 @@ public class NotificationImportService : INotificationImportService
                 return importResult;
             }
 
+            var hasAnyRecordAlreadyBeenImported = false;
+            foreach (var notification in notifications)
+            {
+                // Check that record hasn't already been imported
+                var foundLtbr = await _notificationRepository.NotificationWithLegacyIdExistsAsync(notification.LTBRID);
+                var foundEts = await _notificationRepository.NotificationWithLegacyIdExistsAsync(notification.ETSID);
```

Comment:
We should probably check whether there is an ID before checking whether if there's a matching notification. So something like:
```
var foundLtbr = notification.LTBRID is not null && (await _notificationRepository.NotificationWithLegacyIdExistsAsync(notification.LTBRID));
``` 
or maybe 
```
var foundLtbr = notification.LTBRID is not null
    ? false
    : await _notificationRepository.NotificationWithLegacyIdExistsAsync(notification.LTBRID);
```
to avoid in-lining the await.